docs/api.json

```diff
@@ -0,0 +1,1070 @@
+        "response": "types.volunteerUser"
+        },
+        "DELETE": {
+          "description": "Delete your own volunteer user account",
+          "auth": true,
+          "intendedFor": ["volunteer"],
+          "scope": ["user_details-own:delete"],
+          "body": {
+            "userName": "String"
+          },
+          "response": null
+        }
+      },
+      "/visitors/me": {
+        "GET": {
+          "description": "Retrieve details of your own user account",
+          "auth": true,
+          "intendedFor": ["visitor"],
+          "scope": ["user_details-own:read"],
+          "query": {
+            "fields": ["String"]
+          },
+          "response": "types.visitorUser"
+        },
+        "PUT": {
+        "description": "Update details of a your own visitor user account",
+        "auth": true,
+        "intendedFor": ["visitor"],
+        "scope": ["user_details-own:write"],
+        "query": {
+          "name": "String",
+          "regionId": "Number",
+          "sectorId": "Number",
+          "organisationId": "Number",
+          "gender": "String",
+          "birthYear": "String",
+          "email": "String",
+          "phoneNumber": "String",
+          "emailConsent": "Boolean",
+          "smsConsent": "Boolean",
+          "qrCode": "String",
+          "delete": "Boolean"
+        },
+        "response": "types.visitorUser"
+        },
+        "DELETE": {
+          "description": "Delete your own visitor user account",
+          "auth": true,
+          "intendedFor": ["visitor"],
+          "scope": ["user_details-own:delete"],
+          "body": {
+            "userName": "String"
+          },
+          "response": null
+        }
+      },
+      "/:id/visit_activities/:id": {
+      "POST": {
+        "description": "Adds a visit for a visitor",
+        "auth": true,
+        "intendedFor": ["visitor"],
+        "scope": ["visit_logs-parent:write"],
+        "body": {
+          "userId": "Number",
+          "activityId": "Number"
+        },
+        "response": "types.visit"
+      }
+      },
+      "/:id/volunteer_logs": {
+        "GET": {
+          "description": "Retreive volunteer information for a user",
+          "auth": true,
+          "intendedFor": ["twineAdmin", "fundingBody", "cbAdmin", "volunteer"],
+          "scope": ["volunteer_logs-child:read", "volunteer_logs-own:read"],
+          "query": {
+            "fields": ["String"],
+            "filter": {
+              "since": "String",
+              "until": "String"
+            }
+          },
+          "response": {
+            "totalVolunteerHours": "Number",
+            "volunteerLogs": ["types.volunteerLog"]
+          }
+        }
+      },
+      "/:me/volunteer_logs": {
+        "POST": {
+          "description": "Add volunteer log for your user account",
+          "auth": true,
+          "intendedFor": ["volunteer"],
+          "scope": ["volunteer_logs-parent:write"],
+          "query": {
```

Comment:
We should be consistent in the use of `query` to describe query strings on get requests and `body` to refer to request bodies.